src/base/bittorrent/session.cpp

```diff
@@ -1837,13 +1837,16 @@ bool Session::deleteTorrent(const QString &hash, bool deleteLocalFiles)
         QString rootPath = torrent->rootPath(true);
         if (!rootPath.isEmpty())
             // torrent with root folder
-            m_savePathsToRemove[torrent->hash()] = rootPath;
+            m_removedTorrents[torrent->hash()] = {torrent->name(), rootPath, true};
         else if (torrent->useTempPath())
             // torrent without root folder still has it in its temporary save path
-            m_savePathsToRemove[torrent->hash()] = torrent->savePath(true);
+            m_removedTorrents[torrent->hash()] = {torrent->name(), torrent->savePath(true), true};
+        else
+            m_removedTorrents[torrent->hash()] = {torrent->name(), "", true};
```

Comment:
What does mean empty `savePathToRemove` and `deletesFiles` set tot true?

Reply:
empty path will not smart remove leftover files
deletesFiles to true: Just a flag for `torrent_removed_alert` handler to not remove the item from the QHash.
When deleting files `torrent_removed_alert` is first fired followed by `torrent_deleted_alert` or `torrent_deleted_alert` depending on what happened to the files.

Reply:
Thanks for the clarification. Then I must say that the names for the flag and the structure are poor.  I would recommend to highlight in the structure name that deletion is in progress (PurgingTorrentData?) and in the flag name say that it means request to libtorrent (fileRemovingRequested?).

P.S. Declaring `m_removedTorrents` as `mutable` would also add readability, IMO.

Reply:
Yes I admit that `RemovedTorrentData` as struct name is poor. We need a verb form that shows progress or continuity. But my English fail me at this point.

Since `mutable` isn't needed I don't want to use it.

Reply:
Maybe, RemovingTorrentData?
Need ask @LordNyriox.

Reply:
@glassez, @evsh, @sledgehammer999:  

>Maybe, `RemovingTorrentData`? 

Yes, `RemovingTorrentData` does seem easier to read than `PurgingTorrentData`.  

…Unless, of course, you **intend** to imply that you are clearing the data from a cache or memory rather than the disk—which the word `Purging` does?

Reply:
Yes, I did. Perhaps I did understand the code incorrectly.

Reply:
@evsh:  Thank you for the clarification.  

Since I am out of my depth with this particular code, I will leave the others to the clarification.  

@glassez, @sledgehammer999:  Does this code remove the torrent from the hard-disk or from memory?

Reply:
@LordNyriox The currently named `RemovedTorrentData` struct is used to hold certain pieces of info of torrents that we removed from the GUI(transfer list) and are we waiting confirmation of the removal from the session from libtorrent. Once confirmation is received we use that data to print log entries about the now completely removed torrent.